tasks/setup.py

```diff
@@ -277,6 +266,353 @@ def setupPulumiConfig(config):
         warn(f"Expecting one of [true, false], got {logs_to_std_err}")
 
 
+def resolve_keypair_opts(
+    config: Config,
+    keypair_name: Optional[str] = None,
+    key_type: Optional[str] = None,
+    key_format: Optional[str] = None,
+    private_key_path: Optional[Path | str] = None,
+    public_key_path: Optional[Path | str] = None,
+    require_key_type: Optional[bool] = False,
+    require_keyfile_exists: Optional[bool] = False,
+):
+    """
+    Resolve key pair options, in decreasing order of priority:
+    - user input (parameters)
+    - local config
+    - defaults
+
+    Returns a dict with the resolved values.
+    """
+    if config.configParams is None or config.configParams.aws is None:
+        raise Exit("Config is missing aws section")
+    awsConf = config.configParams.aws
+
+    # defaults
+    if awsConf.keyPairName:
+        default_keypair_name = awsConf.keyPairName
+    else:
+        default_keypair_name = getpass.getuser()
+    if awsConf.privateKeyPath:
+        default_private_key_path = awsConf.privateKeyPath
+    else:
+        default_private_key_path = None
+    if awsConf.publicKeyPath:
+        default_public_key_path = awsConf.publicKeyPath
+    else:
+        default_public_key_path = None
+
+    # ask for missing values
+    if not keypair_name:
+        keypair_name = ask(f"🔑 Key pair name (default: {default_keypair_name}): ")
+        if not keypair_name:
+            keypair_name = default_keypair_name
+    if not key_type and require_key_type:
+        warn('Creating Windows VMs requires "rsa" key type')
+        keytype_default = "rsa"
+        while True:
+            supported = "|".join(supported_key_types)
+            key_type = ask(f"🔑 Key type [{supported}] (default: {keytype_default}): ")
+            if not key_type:
+                key_type = keytype_default
+            if key_type in supported_key_types:
+                break
+            warn(f"{key_type} is not a supported key type")
+    if not key_format:
+        # Just use pem for now
+        key_format = "pem"
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
What other formats are supported?

Reply:
putty's ppk format